### PR TITLE
Add an ability to pin comments

### LINKED ISSUE #5138: Add the option to mods to pin/highlight comments
Pin/highlight comments (whatever you wanna call it) was on disqus and helped mods to give a short note, redirect people to other places as needed, etc. So it can be helpful for mods to give a notice about whats happening in a comment section for everyone to look at.

### PR DESCRIPTION
**Pull request by LiquidPL**
Resolves #5138.

### CODE REVIEW

File: app/Transformers/CommentTransformer.php [hunk 1]
```diff
@@ -53,6 +53,8 @@ public function transform(Comment $comment)
 
             'edited_at' => json_time($comment->edited_at),
             'edited_by_id' => $comment->edited_by_id,
+
```
Comment (notbakaneko) [inline app/Transformers/CommentTransformer.php]:
```suggestion
```

File: routes/web.php [hunk 2]
```diff
@@ -98,6 +98,9 @@
 Route::post('comments/{comment}/vote', 'CommentsController@voteStore')->name('comments.vote');
 Route::delete('comments/{comment}/vote', 'CommentsController@voteDestroy');
 
+Route::post('comments/{comment}/setPinned', 'CommentsController@setPinned')->name('comments.pin');
```
Comment (notbakaneko) [inline routes/web.php]:
```suggestion
Route::post('comments/{comment}/pin', 'CommentsController@pin')->name('comments.pin');
```

File: routes/web.php [hunk 3]
```diff
@@ -98,6 +98,9 @@
...
+Route::delete('comments/{comment}/setPinned', 'CommentsController@setPinned');
```
Comment (notbakaneko) [inline routes/web.php]:
```suggestion
Route::delete('comments/{comment}/pin', 'CommentsController@pin');
```

File: app/Http/Controllers/CommentsController.php [hunk 4]
```diff
@@ -250,6 +251,27 @@ public function update($id)
         return CommentBundle::forComment($comment)->toArray();
     }
 
+    public function setPinned($id)
```
Comment (notbakaneko) [inline app/Http/Controllers/CommentsController.php]:
```suggestion
    public function pin($id)
```

File: app/Libraries/OsuAuthorize.php [hunk 5]
```diff
@@ -972,6 +972,17 @@ public function checkCommentVote(?User $user, Comment $comment) : string
         return 'ok';
     }
 
+    public function checkCommentPin(?User $user, Comment $comment) : string
+    {
+        $this->ensureLoggedIn($user);
+
+        if ($user->isAdmin()) {
+            return 'ok';
+        }
```
Comment (venix12) [inline app/Libraries/OsuAuthorize.php]: seems like this isn't needed as ``doCheckUser()`` always returns ``ok`` if user is admin

File: resources/assets/lib/models/comment.ts [hunk 6]
```diff
@@ -76,6 +78,11 @@ export class Comment {
     return currentUser.is_admin || currentUser.can_moderate;
   }
 
+  @computed
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/assets/lib/models/comment.ts]: it sure does; also seems pointless

**Pull request merged**